contracts/Witch.sol

```diff
@@ -13,198 +13,358 @@ import "@yield-protocol/utils-v2/contracts/math/WDivUp.sol";
 import "@yield-protocol/utils-v2/contracts/cast/CastU256U128.sol";
 import "@yield-protocol/utils-v2/contracts/cast/CastU256U32.sol";
 
-
-contract Witch is AccessControl() {
+/// @title  The Witch is a Auction/Liquidation Engine for the Yield protocol
+/// @notice The Witch grabs uncollateralized vaults, replacing the owner by itself. Then it sells
+/// the vault collateral in exchange for underlying to pay its debt. The amount of collateral
+/// given increases over time, until it offers to sell all the collateral for underlying to pay
+/// all the debt. The auction is held open at the final price indefinitely.
+/// @dev After the debt is settled, the Witch returns the vault to its original owner.
+contract Witch is AccessControl {
     using WMul for uint256;
     using WMulUp for uint256;
     using WDiv for uint256;
     using WDivUp for uint256;
     using CastU256U128 for uint256;
     using CastU256U32 for uint256;
 
-    event Point(bytes32 indexed param, address value);
-    event IlkSet(bytes6 indexed ilkId, uint32 duration, uint64 initialOffer, uint96 line, uint24 dust, uint8 dec);
-    event Bought(bytes12 indexed vaultId, address indexed buyer, uint256 ink, uint256 art);
     event Auctioned(bytes12 indexed vaultId, uint256 indexed start);
+    event Bought(bytes12 indexed vaultId, address indexed buyer, uint256 ink, uint256 art);
+    event LineSet(bytes6 indexed ilkId, bytes6 baseId, uint32 duration, uint64 proportion, uint64 initialOffer);
+    event LimitSet(bytes6 indexed ilkId, bytes6 baseId, uint96 max, uint24 dust, uint8 dec);
```

Comment:
Maybe is just me, but I find it easier to think on terms of base -> collateral, so I'd be tempted to flip the first 2 values. Also I think the base should be indexed too

Reply:
Me too, but all across the protocol is ink first, art/base afterwards. We took that from MakerDAO in v1, and we have kept with it since. Better not to change it now, unfortunately.